Source/WebCore/platform/graphics/GraphicsContext.cpp

```diff
@@ -81,6 +81,23 @@ void GraphicsContext::restore()
         m_stack.clear();
 }
 
+void GraphicsContext::addFilterEffectStyles(const Vector<FilterEffectStyle> styles)
+{
+    for (auto it = styles.rbegin(); it != styles.rend(); ++it) {
+        save();
```

Comment:
I don't think the `save()` and the `beginTransparencyLayer()` should live here in `GraphicsContext`. It should be the responsibility of the caller to do those things. Setting a filter style should behave exactly like setting other state like fillColor.

Reply:
Done. This function is moved to Filter::applyStyles().